### PR TITLE
Automate log out check

### PR DESCRIPTION
**Pull request by amurKontur**
## Summary by CodeRabbit

- **Refactor**
	- Improved the organization and readability of the login test script by centralizing login actions within the `PageManager` class.
- **New Features**
	- Introduced a `HelperBase` class for improved page interaction in Playwright tests.
- **Tests**
	- Optimized test execution durations by adjusting global timeout settings in Playwright configurations.
- **Chores**
	- Updated environment variables to align with current testing requirements.

### CODE REVIEW

File: e2e/page-objects/loginPage.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+   * This method checks the Log in button visibility and then fills in email and password with special speed emulating real user typing
+   * @param email - should be email in form of string
+   * @param password - should be password in form of string
+   * @param speed - delay between keyboard clicks in ms. High numbers increase test execution!
+   */
+
+  async typeLoginPasswordAndLogin(email: string, password: string, speed: number) {
+    await expect(this.page.getByRole('button', { name: 'Log in' })).toBeVisible();
+    // Getting email field and type in like real user
+    const emailInput = this.page.getByRole('textbox').first();
```
Comment (dqunbp) [inline e2e/page-objects/loginPage.ts:19]: Can we locate email input same way as password?

Comment (amurKontur) [reply e2e/page-objects/loginPage.ts:19]: the same way - not. some other ways - yes. 

Monosnap Disaster Ninja 2024-04-18 18-34-44

File: e2e/page-objects/profilePage.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+export class ProfilePage extends HelperBase {
+  /**
+   * By default, this method checks that at profile page there is a logout button. Also checks that at least one element with "Profile" text is present. After that it compares the email got with email passed to it. But also it can check that no 'Profile' texts and logout buttons are present
+   * @param email expected email to what value should be equal to
+   * @param isVisible true to check profile presence, false - profile absence
+   */
+
+  async checkProfilePageForLogoutBtnAndProfileTitleAndCheckEmailIfNeeded(
+    email: string = 'email',
+    isVisible: boolean = true,
```
Comment (dqunbp) [inline e2e/page-objects/profilePage.ts]: It better to simplify the method, with removing isVisible flag

```ts
  async checkProfilePageForLogoutBtnAndProfileTitleAndCheckEmail(email: string) {
    const logoutBtn = this.page.getByRole('button', { name: 'Logout' });
    const profileTextElements = this.page.getByText('Profile').nth(1);

    await expect(logoutBtn).toBeVisible();
    await expect(profileTextElements).toBeVisible();
    const actualEmail = await this.getProfileEmailValueAndCheckFieldIsDisabled();
    expect(actualEmail).toEqual(email);
  }
  ```

Comment (amurKontur) [reply e2e/page-objects/profilePage.ts]: done

**Review by dqunbp — APPROVED** [review]

**Pull request merged**